### PR TITLE
URLGetter py2 upgrades

### PR DESCRIPTION
**Pull request by MichalMMac**
Changes to the `URLGetter` and its subclasses to enable `autopkg/recipes` py23 compatibiilty upgrade with use of `URLGetter`.

Similiar changes future ported to py2-to-3 branch in PR #584

Changes:

- Oriignal `download()` method was renamed to `download_with_curl()` so we can use **download** for simpler uses. b1e449b
- `URLGetter` now has `prepare_curl_cmd()` with *default* `curl` command e45ce95. This is used by new `download()` method and subclasses. 71296de
- We now use inherited methods instead of calling `super()` everywhere. 3a17705
- New simple `download()` method 32267dc for use cases where we don't want to specify `curl` command. It takes http headers too 7b47aa1. We don't use `text` argument in python2 code but it's there for recipe compatibility with python3 version.

### CODE REVIEW

**Comment by nmcspadden** [comment]
new_urlgetter was merged into master, so please rebase your branch on master (because it's currently out of date, from what I can tell). I thought I deleted the new_urlgetter branch entirely on the github remote, but I apparently did not do so successfully.

**Comment by MichalMMac** [comment]
`new_urlgetter` branch pointed to the same as the `master`. Target now changed to `master`.

File: Code/autopkglib/URLGetter.py [hunk 1]
```diff
@@ -62,10 +62,19 @@ def curl_binary(self):
...
 
+    def prepare_curl_cmd(self):
+        """Assemble basic curl command and return it."""
+        return [self.curl_binary(), "--compressed", "--location"]
+
+    def add_curl_headers(self, curl_cmd, headers):
+        """Add headers to curl_cmd"""
+        if headers:
+            for header, value in headers.items():
+                curl_cmd.extend(["--header", "%s: %s" % (header, value)])
```
Comment (nmcspadden) [inline Code/autopkglib/URLGetter.py]: Use `.format()` instead of string interpolation here, because it'll be easier to transition in the future.

Comment (MichalMMac) [reply Code/autopkglib/URLGetter.py]: Ok. Will do.

Comment (MichalMMac) [reply Code/autopkglib/URLGetter.py]: Pushed 5 new commits replacing string interpolation with format in `URLGetter` and subclasses.

**Comment by nmcspadden** [comment]
I tested this and it works for all recipes currently in the Py2-to-3 autopkg-recipes branch (as of commit 25518b9b761f943d4c567fcb0944561b32ca3b89)

**Pull request merged**